addons/account/models/account_move.py

```diff
@@ -133,10 +134,11 @@ def _get_default_invoice_incoterm(self):
         help='If this checkbox is ticked, it means that the user was not sure of all the related informations at the time of the creation of the move and that the move needs to be checked again.')
     journal_id = fields.Many2one('account.journal', string='Journal', required=True, readonly=True,
         states={'draft': [('readonly', False)]},
-        domain="[('company_id', '=', company_id)]",
+        check_company=True, domain="[('id', 'in', suitable_journal_ids)]",
         default=_get_default_journal)
+    suitable_journal_ids = fields.Many2many('account.journal', compute='_compute_suitable_journal_ids')
     company_id = fields.Many2one(string='Company', store=True, readonly=True,
-        related='journal_id.company_id', change_default=True)
+        related='journal_id.company_id', change_default=True, default=lambda self: self.env.company)
```

Comment:
Gné? related, readonly, with a default value?

Reply:
yes, because without default value, there's no possibility to use it in domains until the record is saved